src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTab.java

```
@@ -0,0 +1,310 @@
+    }
+
+    /**
+     * Returns the index relative to the gui this tab was registered too
+     */
+    public int getTabIndex()
+    {
+        return tabRegistry.get(parentContainer).indexOf(this);
+    }
+
+    /**
+     * Whether this tab is on the top row or not
+     */
+    public boolean isTopRow()
+    {
+        return getTabIndex() > 11 ? ((getTabIndex() - 12) % 12) < 6 : getTabIndex() < 6;
+    }
+
+    /**
+     * Returns index % 6 to determine column
+     */
+    public int getColumn()
+    {
+        return getTabIndex() > 11 ? ((getTabIndex() - 12) % 12) % 6 : getTabIndex() % 6;
+    }
+
+    /**
+     * @return Whether this tab's parent GUI is the current GUI
+     */
+    public boolean isActiveTab(Class<? extends GuiContainer> guiContainer)
+    {
+        return guiContainer != null && guiContainer.equals(getTargetGui());
+    }
+
+    /**
+     * @return The tabindex / 12
+     */
+    public int getTabPage()
+    {
+        return getTabIndex() / 12;
+    }
+
+    /**
+     * @return The localized name for this tab
+     */
+    public String getLocalizedName()
+    {
+        return "guiGroup." + name;
+    }
+
+    public ResourceLocation getIconResLoc()
+    {
+        return iconResLoc;
+    }
+
+    public ItemStack getIconStack()
+    {
+        return iconStack;
+    }
+
+    /**
+     * @return A list of the appended tabs for the given GuiContainer
+     */
+    public static List<GuiTab> getTabsForGui(Class<? extends GuiContainer> guiContainer)
+    {
+        return tabRegistry.get(guiContainer);
+    }
+
+    /**
+     * Adds the GuiTab to GuiContainer
+     *
+     * @param parentContainer The GuiContainer to add a tab too
+     * @return The GuiTab
+     */
+    public GuiTab addTo(Class<? extends GuiContainer> parentContainer)
+    {
+        if (parentContainer.equals(GuiContainerCreative.class))
+        {
+            FMLLog.log.warn("Cannot add tabs to creative inventory. Use CreativeTabs instead");
```

This should throw an exception instead of logging.